File: src/main/java/uk/co/eelpieconsulting/feedlistener/UrlBuilder.java
package uk.co.eelpieconsulting.feedlistener;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import uk.co.eelpieconsulting.feedlistener.model.Channel;
import uk.co.eelpieconsulting.feedlistener.model.Subscription;

@Component
public class UrlBuilder {

    @Value("${base.url}")
    private String baseUrl;

    public String getBaseUrl() {
        return baseUrl;
    }

    public String getTwitterCallback(String username) {
        return getBaseUrl() + "/twitter/callback/" + username;
    }

    public String getSubscriptionUrl(Subscription subscription) {
        return getBaseUrl() + "/ui/subscriptions/" + subscription.getId();
    }

    public String getSubscriptionUrl(String subscriptionId) {
        return getBaseUrl() + "/ui/subscriptions/" + subscriptionId;
    }

    public String getDeleteSubscriptionUrl(Subscription subscription) {
        return subscriptionUrl(subscription) + "/delete";
    }

    public String getReadSubscriptionUrl(Subscription subscription) {
        return "/ui/subscriptions/" + subscription.id + "/read";
    }

    public String getSubscriptionItemsUrl(Subscription subscription) {
        return subscriptionUrl(subscription) + "/items";
    }

    public String getChannelItemsUrl(Channel channel) {
        return getBaseUrl() + "/channels/" + channel.id + "/items";
    }

    public String getChannelNewSubscriptionUrl(Channel channel) {
        return getBaseUrl() + "/ui/subscriptions/" + channel.id + "/new";
    }

    public String getChannelUrl(Channel channel) {
        return getBaseUrl() + "/ui/channels/" + channel.getId();
    }

    public String getUserUrl() {
        return getBaseUrl() + "/";
    }

    public String getNewChannelUrl() {
        return getBaseUrl() + "/ui/channels/new";
    }

    private String subscriptionUrl(Subscription subscription) {
        return getBaseUrl() + "/subscriptions/" + subscription.getId();
    }
}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/UrlBuilder.kt
package uk.co.eelpieconsulting.feedlistener

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import uk.co.eelpieconsulting.feedlistener.model.Channel
import uk.co.eelpieconsulting.feedlistener.model.Subscription

@Component
class UrlBuilder @Autowired constructor(@Value("\${base.url}") private val baseUrl: String) {

    fun getSubscriptionUrl(subscription: Subscription): String {
        return baseUrl + "/ui/subscriptions/" + subscription.id
    }

    fun getSubscriptionUrl(subscriptionId: String): String {
        return "$baseUrl/ui/subscriptions/$subscriptionId"
    }

    fun getDeleteSubscriptionUrl(subscription: Subscription): String {
        return subscriptionUrl(subscription) + "/delete"
    }

    fun getReadSubscriptionUrl(subscription: Subscription): String {
        return "/ui/subscriptions/" + subscription.id + "/read"
    }

    fun getSubscriptionItemsUrl(subscription: Subscription): String {
        return subscriptionUrl(subscription) + "/items"
    }

    fun getChannelItemsUrl(channel: Channel): String {
        return baseUrl + "/channels/" + channel.id + "/items"
    }

    fun getChannelNewSubscriptionUrl(channel: Channel): String {
        return baseUrl + "/ui/subscriptions/" + channel.id + "/new"
    }

    fun getChannelUrl(channel: Channel): String {
        return baseUrl + "/ui/channels/" + channel.id
    }

    val userUrl: String
        get() = "$baseUrl/"
    val newChannelUrl: String
        get() = "$baseUrl/ui/channels/new"

    private fun subscriptionUrl(subscription: Subscription): String {
        return baseUrl + "/subscriptions/" + subscription.id
    }

}
